packages/runner-ct/src/app/RunnerCt.tsx

```diff
@@ -199,16 +200,37 @@ const RunnerCt = namedObserver('RunnerCt',
           onDragFinished={persistWidth('ctSpecListWidth')}
           onChange={debounce(updateSpecListWidth)}
         >
-          <SpecList
-            specs={props.state.specs}
-            selectedFile={state.spec ? state.spec.relative : undefined}
-            focusSpecList={focusSpecsList}
-            searchRef={searchRef}
-            className={cs(styles.specsList, {
-              'display-none': hideSpecsListIfNecessary(state),
-            })}
-            onFileClick={runSpec}
-          />
+          {
```

Comment:
I'm interested in the team's opinion on this style, especially @dmtrKovalenko and @agg23 who have worked on very big React code-bases. In my past experience we tried to keep the `return` function simple and logic free, instead opting for a variable. Eg:

```tsx
const specListContent = state.specs.length < 1 
  ? <NoSpec />
  : <SpecList />

// ...

return (
  <div>
    {specListContent}
  </div>
)
```

I don't have a super strong opinion, while I do personally prefer this style it's not really a big deal either way. What do you guys think?

Reply:
I will prefer inline option because it takes some time to understand what exactly rendered inside div, especially if jsx is pretty big and you need to go to definition to view specListContent.

Reply:
We can reduce amount of inlined jsx by making an element before return. 

in order to make

```js
state.specs.length < 1 
  ? createNewSpecFiller
  : <SpecsList /> 
```

Reply:
Agreed with Dmitriy. JSX can be quite hard to parse without inline flow control, and in my opinion the flow control is neat enough that it's easily understood.